### DISCUSSION TITLE
Upload/Download progress for single files

### DISCUSSION THREAD

**Discussion by YoSev** [juicedata/juicefs#713] [Ideas]
We have a workflow that triggers additional steps, once a file has been fully uploaded to the object storage.
What would be the best place to integrate such a logic?

It's about exposing file informations like cache stats, upload/download progress etc

**Comment by davies** [comment]
Assuming you don't mount with `--writeback`,  once you close a file, it should be fully uploaded to object storage.

There are a few ways to find the stats:  cat /jfs/.stats, or pull the metrics into Prometheous: https://github.com/juicedata/juicefs/blob/main/docs/en/p8s_metrics.md
